File: src/test/java/info/kfgodel/dyna/testobjects/ClarkKent.java
package info.kfgodel.dyna.testobjects;

/**
 * This type represents one type view of an object for testing purposes
 * Date: 07/05/19 - 00:09
 */
public interface ClarkKent extends GlassesWearer {

  default void takeGlassesOff(){
    this.setHasGlasses(false);
  }
}

File: src/test/kotlin/info/kfgodel/dyna/objects/ClarkKent.kt
package info.kfgodel.dyna.objects

/**
 * This type represents one type view of an object for testing purposes
 * Date: 07/05/19 - 00:09
 */
interface ClarkKent : GlassesWearer {
  fun takeGlassesOff() {
    this.hasGlasses = false
  }
}
